### PR TITLE
Enable a simple color placeholder for loading cards

### PR DESCRIPTION
**Pull request by MrTimscampi**
**Changes**

As a first step towards getting something more polished down the line, this replaces the transparent colors of cards with our main dark grey.

Note: Using the default colors was tried, but it looked a bit too jarring IMO.

Here is how it looks without images:
image

**Issues**

### CODE REVIEW

**Comment by dmitrylyzo** [comment]
Cards without `button` changed.

new background, with `cardContent-button`
plugins-1

new background, no `cardContent-button`
plugins-2

File: src/themes/dark/theme.css [hunk 1]
```diff
@@ -421,8 +421,7 @@ html {
     color: #4285f4;
 }
 
-.card:focus .cardBox.visualCardBox,
-.card:focus .cardBox:not(.visualCardBox) .cardScalable {
+.card:focus .cardBox.visualCardBox {
```
Comment (dmitrylyzo) [inline src/themes/dark/theme.css:424]: You broke focus for "slow" devices.
`visualCardBox` and non-`visualCardBox` is "Poster/Thumb card" and "Poster/Thumb"
`visualCardBox` tells to make border over whole card (with title).

For testing set some TV user agent.
`Mozilla/5.0 (SMART-TV; Linux; Tizen 4.0) AppleWebKit/537.36 (KHTML, like Gecko) SamsungBrowser/2.1 Chrome/56.0.2924.0 TV Safari/537.36`

Comment (MrTimscampi) [reply src/themes/dark/theme.css:424]: Oh, didn't mean to remove that :/

File: src/components/cardbuilder/card.css [hunk 2]
```diff
@@ -192,37 +192,35 @@ button::-moz-focus-inner {
...
 
 .visualCardBox .cardContent {
     border-bottom-left-radius: 0;
     border-bottom-right-radius: 0;
 }
 
+.backdropCard .cardContent:not(.defaultCardBackground) {
+    background-color: transparent;
+}
+
```
[the remark this answers is not in the document]
Comment (MrTimscampi) [reply src/components/cardbuilder/card.css:223]: For the plugin cards.

[the remark this answers is not in the document]
Comment (dmitrylyzo) [reply src/components/cardbuilder/card.css:223]: I see. This is required for Purple Haze theme.

**Comment by dmitrylyzo** [comment]
> We could try to move background property to `cardScalable`.

Tried on "Purple Haze": round corners, overflow should be moved to `cardScalable` too.

**Comment by MrTimscampi** [comment]
@dmitrylyzo Made a fix using the ::after pseudo-element.

If the image is lazy loaded, it starts as an opaque element of a color defined in the theme. After the image is loaded, the pseudo-element transitions to being transparent, which effectively hides the image's rendering and fades it in.

**Comment by MrTimscampi** [comment]
@dmitrylyzo I reworked this a bit.

It keeps the new lazy loader intact and, instead, does its thing on the padding element underneath the card.

**Comment by dmitrylyzo** [comment]
_Now it is perfect_

Slow TV `Poster`
card-1
Slow TV `Poster card`
card-2
Desktop non-cardBuilder
card-3

<details>
<summary>cardPadder with shadow</summary>

```diff
diff --git a/src/components/cardbuilder/card.css b/src/components/cardbuilder/card.css
index dc11ae915..d77fe5660 100644
--- a/src/components/cardbuilder/card.css
+++ b/src/components/cardbuilder/card.css
@@ -209,6 +209,10 @@ button::-moz-focus-inner {
 }
 
 .cardContent:not(.defaultCardBackground) {
+    background-color: transparent;
+}
+
+.cardBox:not(.visualCardBox) .cardPadder {
     background-color: #242424;
 }
 
@@ -217,7 +221,8 @@ button::-moz-focus-inner {
     border-bottom-right-radius: 0;
 }
 
-.cardContent-shadow {
+.cardContent-shadow,
+.cardBox:not(.visualCardBox) .cardPadder {
     box-shadow: 0 0.0725em 0.29em 0 rgba(0, 0, 0, 0.37);
 }
 
diff --git a/src/components/cardbuilder/cardBuilder.js b/src/components/cardbuilder/cardBuilder.js
index e38f2f75f..0d158e1f4 100644
--- a/src/components/cardbuilder/cardBuilder.js
+++ b/src/components/cardbuilder/cardBuilder.js
@@ -1441,9 +1441,6 @@ import 'programStyles';
             let cardScalableClose = '';
 
             let cardContentClass = 'cardContent';
-            if (!options.cardLayout) {
-                cardContentClass += ' cardContent-shadow';
-            }
 
             if (layoutManager.tv) {
                 // Don't use the IMG tag with safari because it puts a white border around it
@@ -1459,7 +1456,7 @@ import 'programStyles';
 
             let cardScalableClass = 'cardScalable';
 
-            cardImageContainerOpen = '<div class="' + cardBoxClass + '"><div class="' + cardScalableClass + '"><div class="cardPadder-' + shape + '"></div>' + cardImageContainerOpen;
+            cardImageContainerOpen = '<div class="' + cardBoxClass + '"><div class="' + cardScalableClass + '"><div class="cardPadder cardPadder-' + shape + '"></div>' + cardImageContainerOpen;
             cardBoxClose = '</div>';
             cardScalableClose = '</div>';
 
diff --git a/src/themes/appletv/theme.css b/src/themes/appletv/theme.css
index 4b1c91c14..cd0ac8680 100644
--- a/src/themes/appletv/theme.css
+++ b/src/themes/appletv/theme.css
@@ -123,7 +123,7 @@ html {
 
 .paperList,
 .visualCardBox,
-.cardContent:not(.defaultCardBackground) {
+.cardBox:not(.visualCardBox) .cardPadder {
     background-color: rgba(0, 0, 0, 0.1);
 }
 
@@ -464,18 +464,3 @@ html {
 .metadataSidebarIcon {
     color: #00a4dc;
 }
-
-.cardPadder-square,
-.cardPadder-mixedSquare,
-.cardPadder-overflowSquare,
-.cardPadder-portrait,
-.cardPadder-mixedPortrait,
-.cardPadder-overflowPortrait,
-.cardPadder-banner,
-.cardPadder-backdrop,
-.cardPadder-mixedBackdrop,
-.cardPadder-smallBackdrop,
-.cardPadder-overflowBackdrop,
-.cardPadder-overflowSmallBackdrop {
-    background-color: #d2d2d2;
-}
diff --git a/src/themes/blueradiance/theme.css b/src/themes/blueradiance/theme.css
index 16465486c..bc718c2d8 100644
--- a/src/themes/blueradiance/theme.css
+++ b/src/themes/blueradiance/theme.css
@@ -450,7 +450,7 @@ html {
     color: #4285f4;
 }
 
-.cardContent:not(.defaultCardBackground) {
+.cardBox:not(.visualCardBox) .cardPadder {
     background-color: rgba(0, 0, 0, 0.5);
 }
 
@@ -487,18 +487,3 @@ html {
 .metadataSidebarIcon {
     color: #00a4dc;
 }
-
-.cardPadder-square,
-.cardPadder-mixedSquare,
-.cardPadder-overflowSquare,
-.cardPadder-portrait,
-.cardPadder-mixedPortrait,
-.cardPadder-overflowPortrait,
-.cardPadder-banner,
-.cardPadder-backdrop,
-.cardPadder-mixedBackdrop,
-.cardPadder-smallBackdrop,
-.cardPadder-overflowBackdrop,
-.cardPadder-overflowSmallBackdrop {
-    background-color: #022950;
-}
diff --git a/src/themes/dark/theme.css b/src/themes/dark/theme.css
index 69fbb602e..f9163d82f 100644
--- a/src/themes/dark/theme.css
+++ b/src/themes/dark/theme.css
@@ -453,18 +453,3 @@ html {
 .metadataSidebarIcon {
     color: #00a4dc;
 }
-
-.cardPadder-square,
-.cardPadder-mixedSquare,
-.cardPadder-overflowSquare,
-.cardPadder-portrait,
-.cardPadder-mixedPortrait,
-.cardPadder-overflowPortrait,
-.cardPadder-banner,
-.cardPadder-backdrop,
-.cardPadder-mixedBackdrop,
-.cardPadder-smallBackdrop,
-.cardPadder-overflowBackdrop,
-.cardPadder-overflowSmallBackdrop {
-    background-color: #202020;
-}
diff --git a/src/themes/light/theme.css b/src/themes/light/theme.css
index 14c7d5f4a..acea3f1d6 100644
--- a/src/themes/light/theme.css
+++ b/src/themes/light/theme.css
@@ -432,7 +432,7 @@ html {
     color: #4285f4;
 }
 
-.cardContent:not(.defaultCardBackground) {
+.cardBox:not(.visualCardBox) .cardPadder {
     background-color: #fff;
 }
 
@@ -444,18 +444,3 @@ html {
 .metadataSidebarIcon {
     color: #00a4dc;
 }
-
-.cardPadder-square,
-.cardPadder-mixedSquare,
-.cardPadder-overflowSquare,
-.cardPadder-portrait,
-.cardPadder-mixedPortrait,
-.cardPadder-overflowPortrait,
-.cardPadder-banner,
-.cardPadder-backdrop,
-.cardPadder-mixedBackdrop,
-.cardPadder-smallBackdrop,
-.cardPadder-overflowBackdrop,
-.cardPadder-overflowSmallBackdrop {
-    background-color: #fff;
-}
diff --git a/src/themes/purplehaze/theme.css b/src/themes/purplehaze/theme.css
index 1a78e01f7..5262627ed 100644
--- a/src/themes/purplehaze/theme.css
+++ b/src/themes/purplehaze/theme.css
@@ -548,12 +548,18 @@ a[data-role=button] {
 }
 
 .personCard .cardScalable {
-    border-radius: 50%;
+    border-radius: 50% !important;
     border: 1px solid rgb(255, 255, 255);
 }
 
-.cardContent:not(.defaultCardBackground) {
+.cardBox:not(.visualCardBox) .cardPadder {
     background-color: rgba(0, 0, 0, 0.5);
+    border-radius: 1em;
+}
+
+.card.show-focus:not(.show-animation) .cardBox.visualCardBox,
+.card.show-focus:not(.show-animation) .cardBox:not(.visualCardBox) .cardScalable {
+    border-radius: 1.5em;
 }
 
 .card:focus .cardBox.visualCardBox,
@@ -618,23 +624,9 @@ a[data-role=button] {
     right: 20%;
 }
 
-.cardPadder-square,
-.cardPadder-mixedSquare,
-.cardPadder-overflowSquare,
-.cardPadder-portrait,
-.cardPadder-mixedPortrait,
-.cardPadder-overflowPortrait,
-.cardPadder-banner,
-.cardPadder-backdrop,
-.cardPadder-mixedBackdrop,
-.cardPadder-smallBackdrop,
-.cardPadder-overflowBackdrop,
-.cardPadder-overflowSmallBackdrop {
-    background-color: #200c3d;
-}
-
 .personCard .cardPadder-overflowPortrait,
 .personCard .cardPadder-portrait {
     padding-bottom: 100%;
     contain: strict;
+    border-radius: 50% !important;
 }
diff --git a/src/themes/wmc/theme.css b/src/themes/wmc/theme.css
index 0e2ff6f5a..97c08caf8 100644
--- a/src/themes/wmc/theme.css
+++ b/src/themes/wmc/theme.css
@@ -430,7 +430,7 @@ html {
     color: #4285f4;
 }
 
-.cardContent:not(.defaultCardBackground) {
+.cardBox:not(.visualCardBox) .cardPadder {
     background-color: #0f3562;
 }
 
@@ -463,18 +463,3 @@ html {
 .metadataSidebarIcon {
     color: #00a4dc;
 }
-
-.cardPadder-square,
-.cardPadder-mixedSquare,
-.cardPadder-overflowSquare,
-.cardPadder-portrait,
-.cardPadder-mixedPortrait,
-.cardPadder-overflowPortrait,
-.cardPadder-banner,
-.cardPadder-backdrop,
-.cardPadder-mixedBackdrop,
-.cardPadder-smallBackdrop,
-.cardPadder-overflowBackdrop,
-.cardPadder-overflowSmallBackdrop {
-    background-color: #0f3562;
-}
```
</details>

Btw, you left 2 `cardContent-button`: `remotecontrol.js` and `appletv/theme.css`

**Comment by ferferga** [comment]
What are we going to do with this now that we have blurhash? Although it might seem mutually exclusive, I think this is a good idea for having some visual feedback during the small second that it takes for blurhash in large libraries (or when the user has blurhash disabled).

Maybe colors are not a good idea, but what about adding the media type icon as well? [(Example here)](https://github.com/jellyfin/jellyfin-web/pull/909#issuecomment-605913197)

**Pull request merged**